src/Chocolatey.PowerShell/CONTRIBUTING.md

```diff
@@ -0,0 +1,100 @@
+# Contributing Guidelines for `Chocolatey.PowerShell`
+
+This document outlines some guidelines and design practices followed in the `Chocolatey.PowerShell` project.
+It also highlights any important differences from how cmdlets are implemented here compared to the standard patterns for PowerShell cmdlets.
+
+## Naming conventions
+
+Cmdlet classes should be named `VerbNounCommand` and placed in the `Commands` folder and `Chocolatey.PowerShell.Commands` namespace.
+
+## Inherit from `ChocolateyCmdlet` and **not** `PSCmdlet` or `Cmdlet`
+
+`ChocolateyCmdlet` affords some additional helper methods and establishes patterns which can be easily reused across all Chocolatey cmdlets.
+Note that you will still need to apply the standard `[Cmdlet(Verb, Noun)]` attribute on all cmdlet classes for PowerShell to recognise them.
+
+### Overrides
+
+Note that unlike `PSCmdlet`, `ChocolateyCmdlet` requires cmdlets to override the methods `Begin()`, `Process()`, and `End()`. These correspond to `begin {}`, `process {}`, and `end {}` blocks in a PowerShell function.
+
+If unsure, follow these guidelines:
+- Commands that do pipeline processing (declaring a parameter with `[Parameter(ValueFromPipeline = true)]`) will need to handle that input in `Process()`
+- Commands not using pipeline input often handle the bulk of their processing in `End()`
+- If there is any setup that may need to be handled in the command prior to processing any pipeline input, that can go in `Begin()`.
+
+### Logging
+
+Cmdlets inheriting from `ChocolateyCmdlet` will log their parameter values to debug logs when called by default. If any parameters may contain sensitive information, override the `Logging` property and set it to `false` to disable this behaviour.
+
+### Output
+
+By default, `ChocolateyCmdlet`'s `WriteObject()` method will enumerate collections when outputting them, similar to how PowerShell's `Write-Output` works by default. If you need to disable this, use the `WriteObject `
+
+### Helpers
+
+`ChocolateyCmdlet` provides some helper methods for common operations that might be needed for many cmdlets.
+Some of these (and many more) are available on the `PSHelper` class.
```

Comment:
Is there more to this section we want to include in this section? Is the intention that the next section answers the questions about what kind of things might be here? Should that perhaps be mentioned?

Reply:
Mmmm. I think most of that is answered in the next section.... the exact line for what goes onto `ChocolateyCmdlet` _itself_ is a little fuzzier and I don't have a fantastic way to delineate that really... In theory anything that takes `PSCmdlet` as a parameter _could_ be a helper method attached to `ChocolateyCmdlet`, but in practice I think the useful set of things to have directly available is a good bit narrower given calling into `PSHelper` is not really a big ask.